src/device/pf_cmrpc.c

```diff
@@ -2120,49 +2088,54 @@ int pf_cmrpc_rm_ccontrol_req(
       control_io.alarm_sequence_number = 0;  /* Reserved */
       control_io.control_block_properties = 0;
 
-      memset(p_sess->buffer, 0, sizeof(p_sess->buffer));
-      pos = 0;
+      memset(p_sess->out_buffer, 0, sizeof(p_sess->out_buffer));
+      p_sess->out_buf_len = 0;
 
       hdr_pos = 0;
       length_of_body_pos = 0;
 
-      pf_put_dce_rpc_header(&rpc_req, sizeof(p_sess->buffer), p_sess->buffer, &pos, &length_of_body_pos);
+      pf_put_dce_rpc_header(&rpc_req, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len, &length_of_body_pos);
+
+      start_pos = p_sess->out_buf_len;
 
-      start_pos = pos;
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.args_maximum, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
+      hdr_pos = p_sess->out_buf_len;
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.args_length, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.maximum_count, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.offset, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.actual_count, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
 
-      pf_put_uint32(rpc_req.is_big_endian, ndr_data.args_maximum, sizeof(p_sess->buffer), p_sess->buffer, &pos);
-      hdr_pos = pos;
-      pf_put_uint32(rpc_req.is_big_endian, ndr_data.args_length, sizeof(p_sess->buffer), p_sess->buffer, &pos);
-      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.maximum_count, sizeof(p_sess->buffer), p_sess->buffer, &pos);
-      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.offset, sizeof(p_sess->buffer), p_sess->buffer, &pos);
-      pf_put_uint32(rpc_req.is_big_endian, ndr_data.array.actual_count, sizeof(p_sess->buffer), p_sess->buffer, &pos);
+      /* Write the _real_ value for args_maximum */
+      ndr_data.args_maximum = PF_FRAME_BUFFER_SIZE - p_sess->out_buf_len;
+      temp_pos = start_pos;
+      pf_put_uint32(rpc_req.is_big_endian, ndr_data.args_maximum, sizeof(p_sess->out_buffer), p_sess->out_buffer, &temp_pos);
```

Comment:
With thsi new block the ApplicationReady request is rejected by the S7 with "Fault: seq: 0: status: nca_s_fault_remote_no_memory". When this block is inactive, everything works fine.
Only difference I can see is, that without this block ArgsMaximum is 1500, with this block it is 1400.
MaximumCount is 1500 in both cases.